src/empty-fields.js

```diff
@@ -0,0 +1,72 @@
+/**
+ *
+ * @licstart  The following is the entire license notice for the JavaScript code in this file.
+ *
+ * MARC record validators used in Melinda
+ *
+ * Copyright (C) 2014-2018 University Of Helsinki (The National Library Of Finland)
+ *
+ * This file is part of marc-record-validators-melinda
+ *
+ * marc-record-validators-melinda program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * marc-record-validators-melinda is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @licend  The above is the entire license notice
+ * for the JavaScript code in this file.
+ *
+ */
+import 'babel-polyfill';
+import {isEmpty, find} from 'lodash';
+
+export default async function () {
+	return {
+		description: 'Handles empty fields',
+		validate: async record => ({
+			valid: validateRecord(record)
+		}),
+		fix: async record => (
+			record.fields
+				.filter(field => emptyControlFields(field) || emptySubfieldValues(field) || emptySubfields(field))
+				.forEach(field => {
+					if (Object.prototype.hasOwnProperty.call(field, 'subfields') && !isEmpty(field.subfields)) {
```

Comment:
Would be prettier as `'subfields' in field` instead of `Object.prototype.hasOwnProperty.call`

Reply:
Changed